### PR TITLE
only check specific versions for Lantern

### PR DESCRIPTION
**Pull request by joesis**
So beam update can work

### CODE REVIEW

**Comment by joesis** [comment]
The change is deployed to production and both beam and Lantern can be updated, but I'll leave the PR open for reviewing.

**Comment by joesis** [comment]
Hmm still have issue downloading the assets, will look

**Comment by myleshorton** [comment]
Oh darn maybe the file names are wrong?

**Comment by joesis** [comment]
Interestingly the error is like below when I added the asset URL to the error message, but I can get the URL via command line without issue.

> Nov 12 07:02:58.205 - 0m32s ERROR autoupdate: autoupdate.go:90 Patching failed: "Non 2XX response when downloading https://github.com/xiaoshoudian/xiazai/releases/download/0.1.12/update_darwin_amd64.bz2: 404 Not Found"

**Comment by joesis** [comment]
So update succeeds when via the proxy but fails when via DDF. I wonder if the same happens for Lantern as well.

**Comment by joesis** [comment]
Okay this is it. The request is sent to update-server instead of GitHub
```
64.252.106.103 - - [12/Nov/2020:01:57:43 -0500] "GET /xiaoshoudian/xiazai/releases/download/0.1.12/update_windows_386.bz2 HTTP/1.1" 404 19 "-" "Amazon CloudFront"
```

**Comment by joesis** [comment]
Weirdly github.com and update.getlantern.org maps to the same distribution in CloudFront, which is added by https://github.com/getlantern/flashlight/commit/83186f051#diff-e082ee5af8475ee1c113b5e3b1584eaf1585370280dde9c4d30f2e92289b736bR94

**Comment by joesis** [comment]
Finally found out why 😃 

image


Sorry I totally forgot it even though [I reviewed the PR](https://github.com/getlantern/flashlight/pull/471), but @forkner can you by chance remember why we use the same distribution for both, instead of creating a different one? Link to AWS https://console.aws.amazon.com/cloudfront/home?region=ap-southeast-1#distribution-settings:EXPKIP4K8O6FR. I'll update it to direct beam downloads to GitHub for now.

**Comment by joesis** [comment]
But beam still got 404 Not Found. Maybe there's a different problem when downloading via Akamai? I didn't see any 404 on the update-server being from Akamai.

**Comment by joesis** [comment]
Oh well Akamai is because of this. This looks like a case [Terraform can be helpful](https://github.com/getlantern/lantern-internal/issues/3859)

image

File: server/server.go [hunk 1]
```diff
@@ -134,22 +139,8 @@ func (g *ReleaseManager) CheckForUpdate(p *Params) (res *Result, err error) {
...
-	if osVersion, err := semver.Parse(p.OSVersion); err == nil {
-		if p.OS == "windows" && osVersion.LT(semver.MustParse("6.0.0")) {
-			// Windows XP/2003 or below
-			specificVersionToUpgrade = lastVersionForWindowsXP
-		} else if p.OS == "darwin" && osVersion.LT(semver.MustParse("15.0.0")) {
-			// 0SX 10.10 Yosemite or below
-			specificVersionToUpgrade = lastVersionForOSXYosemite
-		} else if p.OS == "linux" {
-			// TODO: remove this once Linux version works again on < Ubuntu 20.04
-			specificVersionToUpgrade = lastVersionForUbuntu20Minus
```
Comment (joesis) [inline server/server.go:148]: This is removed as 6.1.x works on older version of Ubuntu again.

File: server/update_test.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func TestReachServer(t *testing.T) {
 		t.Fatal("Expecting some URL.")
 	}
 
-	res, err = param.CheckForUpdate(fmt.Sprintf("http://%s/update/getlantern/lantern", localAddr), up)
+	res, err = param.CheckForUpdate(fmt.Sprintf("http://%s/update/lantern", localAddr), up)
```
Comment (joesis) [inline server/update_test.go:54]: This was an error inherited from https://github.com/getlantern/autoupdate-server/pull/14

**Comment by myleshorton** [comment]
Wow great sleuthing @joesis! Code LGTM

**Comment by joesis** [comment]
Thanks @myleshorton !

**Pull request merged**